lib/matplotlib/figure.py

```diff
@@ -1142,66 +1168,114 @@ def add_axes(self, *args, **kwargs):
         a.stale_callback = _stale_figure_callback
         return a
 
+    @docstring.dedent_interpd
     def add_subplot(self, *args, **kwargs):
         """
-        Add a subplot.
+        Add a subplot to the figure.
 
         Call signatures::
 
            add_subplot(nrows, ncols, index, **kwargs)
            add_subplot(pos, **kwargs)
+           add_subplot(ax)
 
         Parameters
         ----------
         *args
             Either a 3-digit integer or three separate integers
             describing the position of the subplot. If the three
-            integers are R, C, and P in order, the subplot will take
-            the Pth position on a grid with R rows and C columns.
+            integers are *nrows*, *ncols*, and *index* in order, the
+            subplot will take the *index* position on a grid with *nrows*
+            rows and *ncols* columns. *index* starts at 1 in the upper left
+            corner and increases to the right.
+
+            A 3-digit integer, *pos*, can be used if all integers are less
+            than 10.
 
         projection : {None, 'aitoff', 'hammer', 'lambert', 'mollweide', \
-'polar', rectilinear'}, optional
-            The projection type of the axes.
+'polar', 'rectilinear', str}, optional
+            The projection type of the `~.axes.Axes`. *str* is the name of
```

Comment:
The projection type of the subplot (`~.axes.Axes`).  

(or introduce the fact that this method produces an Axes earlier.  Otherwise the naive user is going to wonder what you are talking about...